packages/nextcloud_test/lib/src/test_client.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:nextcloud/core.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:nextcloud_test/src/docker_container.dart';
+import 'package:process_run/process_run.dart';
+import 'package:universal_io/io.dart';
+
+/// An extension for creating [NextcloudClient]s based on [DockerContainer]s.
+extension TestNextcloudClient on NextcloudClient {
+  /// Creates a new [NextcloudClient] for a given [container] and [username].
+  ///
+  /// It is expected that the password of the user matches the its [username].
+  /// This is the case for the available test docker containers.
+  static Future<NextcloudClient> create(
+    final DockerContainer container, {
+    final String? username = 'user1',
+  }) async {
+    String? appPassword;
+    if (username != null) {
+      final inputStream = StreamController<List<int>>();
+      final process = runExecutableArguments(
+        'docker',
+        [
+          'exec',
+          '-i',
+          container.id,
+          'php',
+          '-f',
+          'occ',
+          'user:add-app-password',
+          username,
+        ],
+        stdin: inputStream.stream,
+      );
+      inputStream.add(utf8.encode(username));
+      await inputStream.close();
+
+      final result = await process;
+      if (result.exitCode != 0) {
+        throw Exception('Failed to run generate app password command\n${result.stderr}\n${result.stdout}');
+      }
+      appPassword = (result.stdout as String).split('\n')[1];
+    }
+
+    final client = NextcloudClient(
+      Uri(
+        scheme: 'http',
+        host: 'localhost',
+        port: container.port,
+      ),
+      loginName: username,
+      password: username,
+      appPassword: appPassword,
+      cookieJar: CookieJar(),
+    );
+
+    var i = 0;
+    while (true) {
+      try {
+        await client.core.getStatus();
+        break;
+      } catch (error) {
+        if (error is HttpException || error is DynamiteApiException) {
+          i++;
+          await Future<void>.delayed(const Duration(milliseconds: 100));
+          if (i >= 300) {
+            throw TimeoutException('Failed to get the status of the Server. $error');
+          }
+        } else {
+          rethrow;
+        }
+      }
+    }
```

Comment:
I don't think that we should catch `DynamiteApiException`s.

Reply:
I'll give it a test to see what exceptions we actually catch here, since the HttpException was only added for apache which we no longer use anywhere.

Reply:
Ok so on my machine the web server starts fast enough to never throw any exceptions. Let's see if GH actions will run into a problem due to being slower hardware. If not we can completely remove this whole while loop and catch block and even save one http request.

Reply:
Yay, no problems in the CI as well. Removed all of this code